pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ClientCachedDao.java

```diff
@@ -0,0 +1,119 @@
+package org.broadinstitute.ddp.db.dao;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+
+import javax.cache.Cache;
+import javax.cache.expiry.Duration;
+
+import org.broadinstitute.ddp.cache.CacheService;
+import org.broadinstitute.ddp.db.dto.ClientDto;
+import org.broadinstitute.ddp.security.StudyClientConfiguration;
+import org.jdbi.v3.core.Handle;
+
+public class ClientCachedDao extends SQLObjectWrapper<ClientDao> implements ClientDao {
+    private static final String CACHE_NAME = "StudyClientConfigCache.byStudyClientConfigCacheKey";
+
+    private class StudyClientConfigCacheKey implements Serializable {
+        String auth0ClientId;
+        String auth0Domain;
+
+        StudyClientConfigCacheKey(String auth0ClientId, String auth0Domain) {
+            this.auth0ClientId = auth0ClientId;
+            this.auth0Domain = auth0Domain;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            StudyClientConfigCacheKey that = (StudyClientConfigCacheKey) o;
+
+            if (auth0ClientId != null ? !auth0ClientId.equals(that.auth0ClientId) : that.auth0ClientId != null) {
+                return false;
+            }
+            return auth0Domain != null ? auth0Domain.equals(that.auth0Domain) : that.auth0Domain == null;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = auth0ClientId != null ? auth0ClientId.hashCode() : 0;
+            result = 31 * result + (auth0Domain != null ? auth0Domain.hashCode() : 0);
+            return result;
+        }
+    }
+
+    private static Cache<StudyClientConfigCacheKey, StudyClientConfiguration> CLIENT_STUDY_CONFIG_CACHE;
+
+    private static void initCache() {
+        if (CLIENT_STUDY_CONFIG_CACHE == null || CLIENT_STUDY_CONFIG_CACHE.isClosed()) {
+            CLIENT_STUDY_CONFIG_CACHE = CacheService.getInstance().getOrCreateCache(CACHE_NAME, new Duration(MINUTES, 10),
+                    ClientCachedDao.class);
+        }
+    }
+
+    public ClientCachedDao(Handle handle) {
+        super(handle, ClientDao.class);
+        initCache();
+    }
+
+    @Override
+    public JdbiClient getClientDao() {
+        return target.getClientDao();
+    }
+
```

Comment:
Most of the methods are just proxy/pass thrus to the JDBI dao that is actually doing the work.
There are a couple of other ways that this could have been done.

I considered very seriously and partially implemented a generic Java proxy class, but thought that it would be easier to commit errors that way.

Another method (in one that I think we should move towards) is to use a dependency injection framework to instantiate the JDBI dao and use annotations as needed to evict stuff. That seemed to heavy a lift right now, but the mechanics of what it would do would be same.